### PR TITLE
Update Cosign v1.12.0 -> v2.0.0

### PR DESCRIPTION
**Pull request by bradbeck**
Update Cosign to v2.0.0

Release: https://github.com/sigstore/cosign/releases/tag/v2.0.0

### CODE REVIEW

File: examples/gradle-pipeline/README.md [hunk 1]
```diff
@@ -37,8 +37,8 @@ fi
 crane ls "$(echo -n ${IMAGE_URL} | sed 's|:[^/]*$||')"
 
 # Verify the image and the attestation.
-cosign verify --key k8s://tekton-chains/signing-secrets "${IMAGE_URL}"
-cosign verify-attestation --type slsaprovenance --key k8s://tekton-chains/signing-secrets "${IMAGE_URL}"
+cosign verify --insecure-ignore-tlog --key k8s://tekton-chains/signing-secrets "${IMAGE_URL}"
```
Comment (pxp928) [inline examples/gradle-pipeline/README.md:40]: `--insecure-ignore-tlog` is there a secure way to do this that we should be doing instead?

**Review by pxp928 — COMMENTED** [review]
LGTM just a question

File: examples/gradle-pipeline/README.md (hunk 1 above)
Comment (bradbeck) [reply examples/gradle-pipeline/README.md:40]: Only if there is a record of the signing in Rekor. At least that's my understanding. If you want to verify "off-line" you have to use that option.

Comment (bradbeck) [reply examples/gradle-pipeline/README.md:40]: If we switched to keyless signing then we could drop the `--insecure-ignore-tlog`.

**Review by pxp928 — APPROVED** [review]

**Pull request merged**